fvm/src/kernel/default.rs

```diff
@@ -110,14 +111,21 @@ where
         let act = state_tree
             .get_actor(addr)?
             .ok_or(anyhow!("state tree doesn't contain actor"))
-            .or_error(ExitCode::ErrIllegalArgument)?;
+            .or_illegal_argument()?;
+
+        if !is_account_actor(&act.code) {
```

Comment:
Not directly related to deletion bug. But we would have previously returned a fatal error here if we attempted to "resolve" a non-account ID address.